### PR TITLE
[dev] AB#23667 - Banner and Z-Index Changes

### PR DESCRIPTION
**Pull request by morethanfire**

### CODE REVIEW

File: src/dataDisplay/banner.jsx [hunk 1]
```diff
@@ -1,43 +1,198 @@
...
+    onAfterClose: undefined,
+    onOpen: undefined,
+    style: undefined,
+    title: undefined,
+    topPosition: undefined,
+    type: undefined,
+};
 
-class Banner extends Component {
+class Banner extends React.Component {
```
Comment (groberts314) [inline src/dataDisplay/banner.jsx:66]: Should it be a `PureComponent`?  Or too much regression risk in making that change?

File: src/dataDisplay/banner.jsx [hunk 2]
```diff
@@ -1,43 +1,198 @@
...
+        } = this.state;
+
+        if (isOpen) {
+            this.onOpen();
+        }
+    }
+
+    componentDidUpdate(prevProps) {
+        const {
+            isOpen: prevIsOpn,
```
Comment (groberts314) [inline src/dataDisplay/banner.jsx:92]: `prevIsOpn` should be `prevIsOpen`.
I kind of like the convention `wasOpen` for the previous one, and keeping `isOpen` (i.e. don't alias it when you destructure it) for the current one, but the `next` and `prev` prefixes works too (and generally, not just for the special case of `isOpen`).

File: src/dataDisplay/banner.jsx (hunk 1 above)
Comment (IlyaRadinsky) [reply src/dataDisplay/banner.jsx:66]: @groberts314  If we want to use the `PureComponent` - no any on-fly objects/functions should be used in `render()` method. But we have such style object, so, not sure that we have to rewrite the `render()` at the moment.

File: src/styles/theme.js [hunk 3]
```diff
@@ -9,6 +9,12 @@ const theme = {
         keys: breakpointsKeys,
         values: breakpointsValues,
     }),
+    zIndex: {
+        drawer: 10000,
+        modal: 11000,
+        prompt: 12000,
+        banner: 13000,
+    },
```
Comment (IlyaRadinsky) [inline src/styles/theme.js:17]: These constants look different than the `variables.scss` constants.

File: src/dataDisplay/banner.jsx (hunk 1 above)
Comment (groberts314) [reply src/dataDisplay/banner.jsx:66]: Roger that.  Thanks for setting me straight! :-)

File: src/styles/theme.js (hunk 3 above)
Comment (morethanfire) [reply src/styles/theme.js:17]: Correct

Comment (morethanfire) [reply src/styles/theme.js:17]: The only difference is prompt and banner. Really, I don't think we should have prompt in there, but for now it is and should be higher than drawer and modal, but lower than banner.

**Comment by groberts314** [comment]
@more answered @IlyaRadinsky 's question, and in order to unblock things and hopefully fix broken things, I think let's go ahead and merge this, despite the 'changes requested' status.

**Pull request merged**